tests/fault_tolerance/test_vllm_health_check.py

```diff
@@ -192,25 +191,59 @@ def test_vllm_health_check_active(request, runtime_services):
                         f"Killing vLLM engine process {{ pid: {child.pid}, cmdline: '{' '.join(cmdline)}' }}"
                     )
+                    vllm_child = child
                     child.kill()
                     break
 
-            time.sleep(2)  # Give some time for the worker to stabilize
-
-            # Step 5: Send a request triggering the handler to shutdown everything.
-            test_response = send_completion_request("How old are you?", 100, timeout=60)
-            logger.error(f"Test request failed: {test_response}")
-
-            # Step 6: Ensure the worker process has been stopped as a result of the EngineDeadError condition.
-            if worker.is_running():
+            # Wait for the vLLM engine process to actually die
+            if vllm_child:
+                try:
+                    vllm_child.wait(timeout=30)
+                    logger.info(
+                        f"vLLM engine process {vllm_child.pid} has been terminated"
+                    )
+                except Exception as e:
+                    logger.error(f"Failed to wait for vLLM engine process to die: {e}")
+                    pytest.fail(f"vLLM engine process did not terminate: {e}")
+
+            # Step 5: Send a request that should fail due to the dead engine.
+            try:
+                test_response = send_completion_request(
+                    "How old are you?", 100, timeout=60
+                )
+                # Verify the response indicates an error (non-2xx status code)
+                if test_response.status_code < 400:
+                    pytest.fail(
+                        f"Expected error response after killing vLLM engine, but got status {test_response.status_code}"
+                    )
+                logger.info(
+                    f"Request correctly failed with status {test_response.status_code}: {test_response.text}"
+                )
+            except requests.exceptions.RequestException as e:
+                # It's also acceptable for the request to raise an exception
+                logger.info(f"Request correctly failed with exception: {e}")
+
+            # Step 6: Wait for the worker process to stop as a result of the EngineDeadError condition.
+            max_wait = 60  # seconds
+            poll_interval = 0.5  # seconds
+            elapsed = 0
+            while elapsed < max_wait:
+                if not worker.is_running():
+                    logger.info(f"Worker stopped after {elapsed:.1f} seconds")
+                    break
+                time.sleep(poll_interval)
+                elapsed += poll_interval
+            else:
+                # Loop completed without break - worker still running
                 pytest.fail(
-                    "Worker should not be running after killing vLLM engine process."
+                    f"Worker should not be running after killing vLLM engine process (waited {max_wait}s)"
                 )
```

Comment:
I wonder if this can simply utilize the `worker.wait(timeout=60)`, similar to the wait introduced above?